crates/uv-interpreter/src/interpreter.rs

```diff
@@ -488,16 +488,15 @@ impl InterpreterInfo {
     /// unless the Python executable changes, so we use the executable's last modified
     /// time as a cache key.
     pub(crate) fn query_cached(executable: &Path, cache: &Cache) -> Result<Self, Error> {
-        let executable_bytes = executable.as_os_str().as_encoded_bytes();
```

Comment:
\cc @BurntSushi - I changed this to just hash the path directly. Not sure if it's equivalent?

Reply:
It looks like it isn't: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=df4e201815ac56439741d3646c18ad70

Reply:
I'm trying to grok why you need the hashes to be the same here. I'm not understanding that park. Like, even if this generates a different hash, that seems okay, it just means you'll get a cache miss? Or is there something deeper here that I'm missing?

Reply:
They don't need to be the same!

Reply:
It was mostly a Chesterton's Fence of trying to understand why we did this in the first place.